io-engine/src/subsys/nvmf/subsystem.rs

```diff
@@ -165,6 +194,169 @@ fn make_sn<T: AsRef<[u8]>>(uuid: T) -> CString {
 }
 
 impl NvmfSubsystem {
+    fn reset_cb(success: bool, ctx: *mut c_void) {
+        if success {
+            info!(
+                "nvmf_event_handler: controller successfully reset {:?}",
+                success
+            );
+        } else {
+            error!("nvmf_event_handler: failed to reset controller ");
+        }
+        done_cb(ctx, success);
+    }
+
+    pub async fn reset_one_child(dev_name: String, ctx: *mut c_void) {
+        let ctrlr = match NVME_CONTROLLERS.lookup_by_name(&dev_name) {
+            Some(ctrlr) => ctrlr,
+            None => {
+                debug!("device name  {dev_name} not a valid NVMe controller");
+                Self::reset_cb(false, ctx);
+                return;
+            }
+        };
+        debug!("nvmf_event_handler: nvme controller {:?} ", ctrlr);
+        let mut ctrlr = ctrlr.lock();
+        if ctrlr.reset(NvmfSubsystem::reset_cb, ctx, false).is_ok() {
+            debug!("nvmf_event_handler: Reset initiated for {:?} ", ctrlr);
+        } else {
+            error!(
+                "nvmf_event_handler: Failed to initiate reset for {:?} ",
+                ctrlr
+            );
+            Self::reset_cb(false, ctx);
```

Comment:
should we capture the actual error in the log? example:
```suggestion
        match ctrlr.reset(NvmfSubsystem::reset_cb, ctx, false) {
            Ok(_) => debug!("nvmf_event_handler: Reset initiated for {:?} ", ctrlr),
            Err(error) => {
                error!(
                    "nvmf_event_handler: Failed to initiate reset for {ctrlr:?} with error: {error:?}"
                );
                Self::reset_cb(false, ctx);
            }
```
Also, in case of error shouldn't the current function return an error?